pydy/system.py

```diff
@@ -340,22 +341,25 @@ def times(self):
         equations of motion are integrated, numerically.
 
         The object should be in a format which the integration module to be
-        used can accept. Since this attribute is not checked for
-        compatibility, the user becomes responsible to supply it correctly.
+        used can accept.
         """
         return self._times
 
     @times.setter
     def times(self, new_times):
-        self._times = new_times
+        self._times = np.asarray(new_times)
+        self._check_times(self._times)
 
     def _check_times(self, times):
```

Comment:
Why is it necessary to pass in times here? Why not just use self.times, or better yet why have this separate function at all? You can just move the if statements into the setter.